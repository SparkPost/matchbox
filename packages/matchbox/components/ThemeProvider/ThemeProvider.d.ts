export default Theme;
/**
 * Provides context for styled-system
 *
 * - https://www.styled-components.com/docs/advanced#theming
 * - https://styled-system.com/theme-specification
 *
 * @prop theme Overrides matchbox's theme
 */
declare function Theme(props: any): any;
declare namespace Theme {
    const displayName: string;
    namespace propTypes {
        const skipGlobalStyles: any;
        const target: any;
    }
}
