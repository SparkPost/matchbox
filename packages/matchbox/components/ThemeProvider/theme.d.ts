export default theme;
declare namespace theme {
    namespace colors {
        const gray: {
            100: any;
            200: any;
            300: any;
            400: any;
            500: any;
            600: any;
            700: any;
            800: any;
            900: any;
            1000: any;
        };
        const blue: {
            100: any;
            200: any;
            300: any;
            400: any;
            500: any;
            600: any;
            700: any;
            800: any;
            900: any;
            1000: any;
        };
        const red: {
            100: any;
            200: any;
            300: any;
            400: any;
            500: any;
            600: any;
            700: any;
            800: any;
            900: any;
            1000: any;
        };
        const yellow: {
            100: any;
            200: any;
            300: any;
            400: any;
            500: any;
            600: any;
            700: any;
            800: any;
            900: any;
            1000: any;
        };
        const green: {
            100: any;
            200: any;
            300: any;
            400: any;
            500: any;
            600: any;
            700: any;
            800: any;
            900: any;
            1000: any;
        };
        const teal: {
            100: any;
            200: any;
            300: any;
            400: any;
            500: any;
            600: any;
            700: any;
            800: any;
            900: any;
            1000: any;
        };
        const purple: {
            100: any;
            200: any;
            300: any;
            400: any;
            500: any;
            600: any;
            700: any;
            800: any;
            900: any;
            1000: any;
        };
        const magenta: {
            100: any;
            200: any;
            300: any;
            400: any;
            500: any;
            600: any;
            700: any;
            800: any;
            900: any;
            1000: any;
        };
        namespace brand {
            export const orange: any;
            const gray_1: any;
            export { gray_1 as gray };
            const blue_1: any;
            export { blue_1 as blue };
        }
        const white: any;
    }
    const space: {
        0: any;
        100: any;
        200: any;
        300: any;
        400: any;
        450: any;
        500: any;
        550: any;
        600: any;
        650: any;
        700: any;
        750: any;
        800: any;
    };
    const sizes: {
        0: any;
        100: any;
        200: any;
        300: any;
        400: any;
        450: any;
        500: any;
        550: any;
        600: any;
        650: any;
        700: any;
        750: any;
        800: any;
        900: any;
        1000: any;
        1100: any;
        1150: any;
        1200: any;
        1300: any;
        1350: any;
        1400: any;
        full: string;
    };
    const radii: {
        0: any;
        100: any;
        200: any;
        pill: any;
        circle: any;
    };
    const borders: {
        300: string;
        400: string;
        500: string;
    };
    const borderWidths: {
        0: any;
        100: any;
        200: any;
    };
    const shadows: {
        0: any;
        100: any;
        200: any;
        300: any;
        400: any;
    };
    namespace fonts {
        const sans: any;
        const monospace: any;
    }
    const fontSizes: {
        50: any;
        100: any;
        200: any;
        300: any;
        400: any;
        500: any;
        600: any;
        700: any;
        800: any;
    };
    const lineHeights: {
        100: any;
        200: any;
        300: any;
        400: any;
        500: any;
        600: any;
        700: any;
        800: any;
    };
    namespace fontWeights {
        const light: any;
        const normal: any;
        const medium: any;
        const semibold: any;
    }
    const breakpoints: any[];
    namespace zIndices {
        export const below: any;
        const _default: any;
        export { _default as default };
        export const overlay: any;
    }
    namespace motion {
        namespace duration {
            export const fast: any;
            const medium_1: any;
            export { medium_1 as medium };
            export const slow: any;
        }
        namespace ease {
            const _in: any;
            export { _in as in };
            export const out: any;
            export const inOut: any;
        }
    }
}
