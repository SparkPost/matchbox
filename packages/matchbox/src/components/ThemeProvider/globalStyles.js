import { tokens } from '@sparkpost/design-tokens';

export default `
  html {
    position: relative;
    box-sizing: border-box;
    font-size: ${tokens.fontSize_root};
  }

  body {
    min-height: 100%;
    margin: 0;
    padding: 0;

    color: ${tokens.color_gray_900};
    font-family: ${tokens.fontFamily_sans};
    font-size: ${tokens.fontSize_300};
    line-height: ${tokens.lineHeight_300};
    font-weight: ${tokens.fontWeight_normal};
  }

  * {
    box-sizing: inherit;
  }

  h1 {
    font-size: ${tokens.fontSize_700};
    line-height: ${tokens.lineHeight_700};
  }

  h2 {
    font-size: ${tokens.fontSize_600};
    line-height: ${tokens.lineHeight_600};
  }

  h3 {
    font-size: ${tokens.fontSize_500};
    line-height: ${tokens.lineHeight_500};
  }

  h4 {
    font-size: ${tokens.fontSize_400};
    line-height: ${tokens.lineHeight_400};
  }

  h5 {
    font-size: ${tokens.fontSize_300};
    line-height: ${tokens.lineHeight_300};
  }

  h6 {
    font-size: ${tokens.fontSize_200};
    line-height: ${tokens.lineHeight_200};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  strong,
  b {
    font-weight: ${tokens.fontWeight_semibold};
  }

  small {
    font-size: ${tokens.fontSize_200};
    line-height: ${tokens.lineHeight_200};
  }

  p,
  ul,
  ol {
    font-size: ${tokens.fontSize_300};
    line-height: ${tokens.lineHeight_300};
  }

  code,
  pre {
    font-family: ${tokens.fontFamily_monospace};
    font-size: ${tokens.fontSize_100};
    line-height: ${tokens.lineHeight_100};
  }

  ul,
  ol,
  dd,
  figure,
  pre,
  table,
  fieldset,
  hr {
    margin-top: 0;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  ul,
  ol,
  pre,
  table,
  fieldset {
    margin: 0;
  }

  a,
  a:visited {
    color: ${tokens.color_blue_700};
    text-decoration: underline;

    &:hover,
    &:active,
    &:focus {
      color: ${tokens.color_blue_800}
      cursor: pointer;
    }
  }

  ul,
  ol {
    padding-left: ${tokens.spacing_500};
  }

  hr {
    margin: ${tokens.spacing_500} 0;
    background: ${tokens.color_gray_400};
    border: none;
    height: 1px;
  }

  ::selection {
    color: ${tokens.color_gray_1000};
    background: ${tokens.color_blue_300};
    text-shadow: none;
  }

  [tabindex="-1"]:focus {
    outline: none;
  }
`;
