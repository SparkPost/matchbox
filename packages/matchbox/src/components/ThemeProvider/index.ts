export { default as ThemeProvider } from './ThemeProvider';
export * from './ThemeProvider';
