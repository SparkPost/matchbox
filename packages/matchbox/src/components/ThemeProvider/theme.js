/* eslint-disable max-lines */
import { tokens } from '@sparkpost/design-tokens';

// Style prop reference:
// https://styled-system.com/table

const theme = {
  // Example: <Box bg="blue.100" color={gray.1000} />
  colors: {
    gray: {
      100: tokens.color_gray_100,
      200: tokens.color_gray_200,
      300: tokens.color_gray_300,
      400: tokens.color_gray_400,
      500: tokens.color_gray_500,
      600: tokens.color_gray_600,
      700: tokens.color_gray_700,
      800: tokens.color_gray_800,
      900: tokens.color_gray_900,
      1000: tokens.color_gray_1000,
    },
    blue: {
      100: tokens.color_blue_100,
      200: tokens.color_blue_200,
      300: tokens.color_blue_300,
      400: tokens.color_blue_400,
      500: tokens.color_blue_500,
      600: tokens.color_blue_600,
      700: tokens.color_blue_700,
      800: tokens.color_blue_800,
      900: tokens.color_blue_900,
      1000: tokens.color_blue_1000,
    },
    red: {
      100: tokens.color_red_100,
      200: tokens.color_red_200,
      300: tokens.color_red_300,
      400: tokens.color_red_400,
      500: tokens.color_red_500,
      600: tokens.color_red_600,
      700: tokens.color_red_700,
      800: tokens.color_red_800,
      900: tokens.color_red_900,
      1000: tokens.color_red_1000,
    },
    yellow: {
      100: tokens.color_yellow_100,
      200: tokens.color_yellow_200,
      300: tokens.color_yellow_300,
      400: tokens.color_yellow_400,
      500: tokens.color_yellow_500,
      600: tokens.color_yellow_600,
      700: tokens.color_yellow_700,
      800: tokens.color_yellow_800,
      900: tokens.color_yellow_900,
      1000: tokens.color_yellow_1000,
    },
    green: {
      100: tokens.color_green_100,
      200: tokens.color_green_200,
      300: tokens.color_green_300,
      400: tokens.color_green_400,
      500: tokens.color_green_500,
      600: tokens.color_green_600,
      700: tokens.color_green_700,
      800: tokens.color_green_800,
      900: tokens.color_green_900,
      1000: tokens.color_green_1000,
    },
    teal: {
      100: tokens.color_teal_100,
      200: tokens.color_teal_200,
      300: tokens.color_teal_300,
      400: tokens.color_teal_400,
      500: tokens.color_teal_500,
      600: tokens.color_teal_600,
      700: tokens.color_teal_700,
      800: tokens.color_teal_800,
      900: tokens.color_teal_900,
      1000: tokens.color_teal_1000,
    },
    purple: {
      100: tokens.color_purple_100,
      200: tokens.color_purple_200,
      300: tokens.color_purple_300,
      400: tokens.color_purple_400,
      500: tokens.color_purple_500,
      600: tokens.color_purple_600,
      700: tokens.color_purple_700,
      800: tokens.color_purple_800,
      900: tokens.color_purple_900,
      1000: tokens.color_purple_1000,
    },
    magenta: {
      100: tokens.color_magenta_100,
      200: tokens.color_magenta_200,
      300: tokens.color_magenta_300,
      400: tokens.color_magenta_400,
      500: tokens.color_magenta_500,
      600: tokens.color_magenta_600,
      700: tokens.color_magenta_700,
      800: tokens.color_magenta_800,
      900: tokens.color_magenta_900,
      1000: tokens.color_magenta_1000,
    },
    brand: {
      orange: tokens.color_brand_orange,
      gray: tokens.color_brand_gray,
      blue: tokens.color_brand_blue,
    },
    white: tokens.color_white,
  },

  // Example: <Box marginBottom="400" padding={400} />
  space: {
    0: tokens.spacing_0,
    100: tokens.spacing_100,
    200: tokens.spacing_200,
    300: tokens.spacing_300,
    400: tokens.spacing_400,
    450: tokens.spacing_450,
    500: tokens.spacing_500,
    550: tokens.spacing_550,
    600: tokens.spacing_600,
    650: tokens.spacing_650,
    700: tokens.spacing_700,
    750: tokens.spacing_750,
    800: tokens.spacing_800,
  },

  // Example: <Box width="1200" height={1/2} />
  sizes: {
    0: tokens.sizing_0,
    100: tokens.sizing_100,
    200: tokens.sizing_200,
    300: tokens.sizing_300,
    400: tokens.sizing_400,
    450: tokens.sizing_450,
    500: tokens.sizing_500,
    550: tokens.sizing_550,
    600: tokens.sizing_600,
    650: tokens.sizing_650,
    700: tokens.sizing_700,
    750: tokens.sizing_750,
    800: tokens.sizing_800,
    900: tokens.sizing_900,
    1000: tokens.sizing_1000,
    1100: tokens.sizing_1100,
    1200: tokens.sizing_1200,
    1300: tokens.sizing_1300,
    1350: tokens.sizing_1350,
    1400: tokens.sizing_1400,
    full: '100%',
  },

  // Example: <Box borderRadius="pill" />
  radii: {
    0: tokens.borderRadius_0,
    100: tokens.borderRadius_100,
    200: tokens.borderRadius_200,
    pill: tokens.borderRadius_pill,
    circle: tokens.borderRadius_circle,
  },

  // Example: <Box border={500} />
  borders: {
    400: `${tokens.borderWidth_100} solid ${tokens.color_gray_400}`,
  },

  // Example: <Box borderWidth={100} />
  borderWidths: {
    0: tokens.borderWidth_0,
    100: tokens.borderWidth_100,
  },

  // Example: <Box boxShadow={200} />
  shadows: {
    0: tokens.boxShadow_0,
    100: tokens.boxShadow_100,
    200: tokens.boxShadow_200,
    300: tokens.boxShadow_300,
    400: tokens.boxShadow_400,
  },

  // Example:
  // <Box
  //   fontFamily="monospace"
  //   fontSize="100"
  //   lineHeight="100"
  //   fontWeight="medium"
  // />
  fonts: {
    sans: tokens.fontFamily_sans,
    monospace: tokens.fontFamily_monospace,
  },

  fontSizes: {
    100: tokens.fontSize_100,
    200: tokens.fontSize_200,
    300: tokens.fontSize_300,
    400: tokens.fontSize_400,
    500: tokens.fontSize_500,
    600: tokens.fontSize_600,
    700: tokens.fontSize_700,
    800: tokens.fontSize_800,
  },

  lineHeights: {
    100: tokens.lineHeight_100,
    200: tokens.lineHeight_200,
    300: tokens.lineHeight_300,
    400: tokens.lineHeight_400,
    500: tokens.lineHeight_500,
    600: tokens.lineHeight_600,
    700: tokens.lineHeight_700,
    800: tokens.lineHeight_800,
  },

  fontWeights: {
    light: tokens.fontWeight_light,
    normal: tokens.fontWeight_normal,
    medium: tokens.fontWeight_medium,
    semibold: tokens.fontWeight_semibold,
  },

  breakpoints: [
    tokens.mediaQuery_xs,
    tokens.mediaQuery_sm,
    tokens.mediaQuery_md,
    tokens.mediaQuery_lg,
    tokens.mediaQuery_xl,
  ],

  zIndices: {
    below: tokens.zIndex_below,
    default: tokens.zIndex_default,
    overlay: tokens.zIndex_overlay,
  },
};

export default theme;
