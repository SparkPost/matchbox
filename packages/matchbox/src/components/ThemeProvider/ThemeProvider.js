import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider, createGlobalStyle, StyleSheetManager } from 'styled-components';
import { normalize } from 'styled-normalize';
import theme from './theme';
import global from './globalStyles';

const GlobalStyle = createGlobalStyle`
  ${normalize}
  ${global}
`;

/**
 * See https://styled-components.com/docs/api#stylesheetmanager
 * Target provides an alternate DOM node to inject styles info
 *
 * Example:
 * In your app's <head>:
 * <style id="styled-components-target"></style>
 *
 * When rendering ThemeProvider:
 * <ThemeProvider target={document.getElementById('styled-components-target')} />
 * */
function Manager({ target, children }) {
  if (!target) {
    return children;
  }

  return <StyleSheetManager target={target}>{children}</StyleSheetManager>;
}

Manager.displayName = 'MatchboxStyleSheetManager';

/**
 * Provides context for styled-system
 *
 * - https://www.styled-components.com/docs/advanced#theming
 * - https://styled-system.com/theme-specification
 *
 * @prop theme Overrides matchbox's theme
 */
function Theme(props) {
  return (
    <Manager>
      <ThemeProvider theme={{ ...theme, ...props.theme }}>
        {/*
          FE-913:`skipGlobalStyles` is to exclude global styles from unit tests.
          The custom assertion `toHaveAttributeValue` fails with this turned on.
          We should either move away from using this assertion (or even enzyme),
          or try to resolve the error
        */}
        {!props.skipGlobalStyles && <GlobalStyle />}

        {props.children}
      </ThemeProvider>
    </Manager>
  );
}

Theme.displayName = 'MatchboxThemeProvider';
Theme.propTypes = {
  skipGlobalStyles: PropTypes.bool,
};

export default Theme;
