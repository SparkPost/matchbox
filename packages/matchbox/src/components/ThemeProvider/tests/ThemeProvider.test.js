import React from 'react';
import { shallow } from 'enzyme';

import ThemeProvider from '../ThemeProvider';

describe('ThemeProvider', () => {
  const subject = (props) => shallow(<ThemeProvider {...props} />);

  it('should be able to modify the theme', () => {
    expect(subject({ theme: { testKey: 'testValue' }}).prop('theme').testKey).toEqual('testValue');
  });
});
