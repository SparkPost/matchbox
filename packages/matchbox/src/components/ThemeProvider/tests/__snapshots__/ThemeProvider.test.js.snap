// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider should use a provided theme 1`] = `
Object {
  "breakpoints": Array [
    "400px",
    "800px",
  ],
  "fontSizes": Object {
    "400": "1rem",
  },
  "lineHeights": Object {
    "400": "1.5rem",
  },
  "mock-theme-key": "mock-theme-value",
  "space": Object {
    "400": "1rem",
    "500": "1.5rem",
  },
}
`;
